tools/dep_updaters/update-libuv.sh

```diff
@@ -2,16 +2,29 @@
 set -e
 # Shell script to update libuv in the source tree to a specific version
 
-BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
-DEPS_DIR="$BASE_DIR/deps"
-LIBUV_VERSION=$1
-
 if [ "$#" -le 0 ]; then
-  echo "Error: please provide an libuv version to update to"
+  echo "Error: please provide an libuv version to update"
   echo "	e.g. $0 1.44.2"
   exit 1
 fi
 
+BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
+DEPS_DIR="$BASE_DIR/deps"
+
+NEW_VERSION=$1
+VERSION_H="$DEPS_DIR/uv/include/uv/version.h"
+CURRENT_MAJOR_VERSION=$(grep "#define UV_VERSION_MAJOR" $VERSION_H | sed -n "s/^.*MAJOR \(.*\)/\1/p")
+CURRENT_MINOR_VERSION=$(grep "#define UV_VERSION_MINOR" $VERSION_H | sed -n "s/^.*MINOR \(.*\)/\1/p")
+CURRENT_PATCH_VERSION=$(grep "#define UV_VERSION_PATCH" $VERSION_H | sed -n "s/^.*PATCH \(.*\)/\1/p")
+CURRENT_SUFFIX_VERSION=$(grep "#define UV_VERSION_SUFFIX" $VERSION_H | sed -n "s/^.*SUFFIX \"\(.*\)\"/\1/p")
```

Comment:
Necessary in case the `nodejs/node` repo was cloned in a folder that has a space in its path.
```suggestion
CURRENT_MAJOR_VERSION=$(grep "#define UV_VERSION_MAJOR" "$VERSION_H" | sed -n "s/^.*MAJOR \(.*\)/\1/p")
CURRENT_MINOR_VERSION=$(grep "#define UV_VERSION_MINOR" "$VERSION_H" | sed -n "s/^.*MINOR \(.*\)/\1/p")
CURRENT_PATCH_VERSION=$(grep "#define UV_VERSION_PATCH" "$VERSION_H" | sed -n "s/^.*PATCH \(.*\)/\1/p")
CURRENT_SUFFIX_VERSION=$(grep "#define UV_VERSION_SUFFIX" "$VERSION_H" | sed -n "s/^.*SUFFIX \"\(.*\)\"/\1/p")
```

Reply:
Good catch ^^ It will help me even in my future script as I never pay too much attention to this :) 🙈